Project-Lily/Project-Lily/DogListViewController.swift

```diff
@@ -96,7 +97,7 @@ extension DogListViewController:  UITableViewDataSource {
         if isSearching == true {
             return filteredDogArray.count
         }
-        return dogDatabase.dogsArray.count
+        return dogListVM.dogs.count
```

Comment:
🍰 The number of rows that this tableView needs to know about could be its own function or computed property. Why is that necessary? It's not hugely important, but it does strive toward the goal of the ViewController not knowing about the Model. The changes to implement that are fairly small, but from the call-site (here) it could look like `dogListVM.numberOfCells`, and line `98` could ultimately look like `dogListVM.numberOfFilteredCells`. I think starting with line 100 will be a simpler task to take on if you decide to make some changes

Reply:
@kendallpoindexter Instead of a function, you could use a computed property to do the same thing and not need the parentheses at the end, as you would with a function. For example:

**With a function**:

```swift
// In your viewModel:
func numberOfCells() -> Int {
    return dogs.count
}

// In your viewController:
func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
    return viewModel.numberOfCells()
}
```

**With a computed property**:

```swift
var numberOfCells: Int {
    return dogs.count
}

// In your viewController:
func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
    return viewModel.numberOfCells
}
```

You can see that the difference is very minor, but it helps keep functions and variables semantically distinct.

Reply:
- [ ] Refactor function into computed property